File: homework-13/frontend/src/services/paymentService.js
import axios from "axios";
import UserService from "./userService";
import authHeader from "./auth-header";
import {API_URL} from "../API_URL";

export const API_URL_PAY = API_URL + "/payment";

const pay = (payment, dispatch) => {

    return axios.post(API_URL_PAY, payment, {headers: authHeader()}).then(
        () => {
            UserService.getUser(payment.userId, dispatch)
        },
        (error) => {
            const _content = (error.response && error.response.data) ||
                error.message ||
                error.toString();

            console.error(_content)
        });
};

const paymentService = {
    pay,
};

export default paymentService

File: homework-14/backend/src/main/java/ru/sber/services/PaymentService.java
package ru.sber.services;

import ru.sber.entities.Payment;

/**
 * Сервис для оплаты товаров
 */
public interface PaymentService {

    /**
     * Совершает платеж
     *
     * @param payment платеж
     * @return boolean
     */
    boolean pay(Payment payment);

}
